src/pages/TestingPage/TestControls/controlReducer.ts

```diff
@@ -0,0 +1,102 @@
+import { Action, Form, FormData } from "./TestAttributes";
+
+const searchId = (form: Form, id: string): number => {
+    return form.formData.findIndex((inputData) => inputData.id == id);
+};
+
+const checkType = (type: string, value: number): boolean => {
+    //TODO: which types?
+    return true;
+};
+
+const checkIsValid = (formData: FormData): boolean => {
+    const result =
+        formData.reduce(
+            (prev, currentInput) =>
+                prev &&
+                (currentInput.validity.isValid || !currentInput.enabled),
+            true
+        ) &&
+        formData.some((input) => input.enabled) &&
+        !formData.some((input) => input.enabled && !input.value);
+    return result;
+};
+
+export const checkValidityInputs = (initialState: Form): Form => {
```

Comment:
What is the difference between checkIsValid and checkValidityInputs

Reply:
checkIsValid (now called isFormValid) checks all the validity.isValid of each inputData to establish the global boolean of the form. checkValidityInputs (now called checkValidityInitialInputs) checks the data of the initialState and establish if each data is correct, setting the variable isValid of each data with the correct state